modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryEnumObjectImpl.java

```diff
@@ -473,4 +473,13 @@ public BinaryEnumObjectImpl(BinaryContext ctx, byte[] arr) {
 
         return reader.afterMessageRead(BinaryEnumObjectImpl.class);
     }
+
+    /** {@inheritDoc} */
+    @Override public int size() {
+        if (valBytes == null) {
```

Comment:
I don't think that returning -1 is meaningful here. We should marshal if size() requested and valBytes is null, such as in valueBytes, e.g.:

        if (valBytes != null)
            return valBytes;

        valBytes = U.marshal(ctx.marshaller(), this);

        return valBytes;

Also, please note that if() with single-line body should not be brace-indented.